homeassistant/components/google_photos/api.py

```diff
@@ -70,6 +72,40 @@ async def list_media_items(
         )
         return await self._execute(cmd)
 
+    async def upload_content(self, content: bytes, mime_type: str) -> str:
```

Comment:
The API also supports resumable uploads which are more reliable and can be retried. It would add a lot of complexity that I'm not sure it's worth it. Maybe something to consider if you extract this to a 3rd party library.

Reply:
Yeah, agreed it would be cool, though maybe not needed.  I could see this being pretty helpful for larger video use cases. I agree that having the 3rd party library implement this could be the way to go, and maybe it could work like a little backup producer/consumer queue.